### PR TITLE
Add HydraError class

### PR DESCRIPTION
**Pull request by vddesai1871**
Fixes #

### Checklist
- [ ] My branch is up-to-date with upstream/develop branch.
- [x] Everything works and tested for Python 3.5.2 and above.

### Description

Added HydraError class as a subclass of HydraStatus to represent errors.

### CODE REVIEW

**Comment by vddesai1871** [comment]
Sorry for this mess! It was because of the master branch having more commits than develop. If I am correct merging this PR and subsequently merging develop to master will sync them.
The important thing is that I would like to have it merged so the changes I am making (using HydraError) at hydrus doesn't fail the integration tests.

File: hydra_python_core/doc_writer.py [hunk 1]
```diff
@@ -519,24 +519,37 @@ def generate(self) -> Dict[str, Any]:
 class HydraStatus():
```
Comment (chrizandr) [inline hydra_python_core/doc_writer.py:519]: Status and error classes are part of the API response, I don't think they belong with the API documentation.
Is doc writer the correct place for this?

Maybe we can have a direct implementation in hydrus?
Any thoughts: @Mec-is @xadahiya ?

**Comment by chrizandr** [comment]
Also you could sqaush the other commits into one before making commits for the new changes. It would make it more clear as to what is new in the PR.

**Comment by vddesai1871** [comment]
Tried it. I am getting an endless list of merge conflicts while rebasing. How come develop is 30 or more commits behind the master branch? Sorry if it's too naive of me, but it seems quite unusual to me.

**Pull request closed without merging**